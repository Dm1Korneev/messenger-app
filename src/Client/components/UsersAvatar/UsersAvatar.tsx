import AccountIcon from '@mui/icons-material/AccountCircleOutlined';
import Avatar from '@mui/material/Avatar';

type Props = {
  avatar?: string;
  author?: string;
  size?: number;
}

export const UsersAvatar = ({ avatar, author, size }: Props) => (
  <Avatar alt={author} src={avatar} sx={{ width: size, height: size }} variant="rounded">
    {!avatar && <AccountIcon fontSize="large" />}
  </Avatar>
);
