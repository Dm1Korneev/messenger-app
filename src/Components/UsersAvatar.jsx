import React from 'react';
import PropTypes from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';

import AccountIcon from '@material-ui/icons/AccountCircleOutlined';

function UsersAvatar(props) {
  const {
    classes, avatar, author, size,
  } = props;

  const style = size && { width: size, height: size };

  return (
    <ListItemAvatar className={classes.ListItemAvatar} style={style}>
      <Avatar alt={author} src={avatar}>
        {!avatar && <AccountIcon style={{ fontSize: 40 }} />}
      </Avatar>
    </ListItemAvatar>
  );
}

UsersAvatar.defaultProps = {
  size: undefined,
  avatar: undefined,
  author: undefined,
};
UsersAvatar.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  avatar: PropTypes.string,
  author: PropTypes.string,
  size: PropTypes.number,
};

const styles = (theme) => ({
  ListItemAvatar: {
    borderRadius: theme.spacing.unit / 2,
  },
});

export default withStyles(styles)(UsersAvatar);
