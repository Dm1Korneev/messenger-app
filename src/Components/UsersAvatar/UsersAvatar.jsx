import React from 'react';
import PropTypes from 'prop-types';
import Avatar from '@material-ui/core/Avatar';
import AccountIcon from '@material-ui/icons/AccountCircleOutlined';

function UsersAvatar(props) {
  const {
    avatar, author, size,
  } = props;

  const style = size ? { width: size, height: size } : {};

  return (
    <Avatar variant="rounded" alt={author} src={avatar} style={style}>
      {!avatar && <AccountIcon style={{ fontSize: 40 }} />}
    </Avatar>
  );
}

UsersAvatar.defaultProps = {
  size: undefined,
  avatar: undefined,
  author: undefined,
};
UsersAvatar.propTypes = {
  avatar: PropTypes.string,
  author: PropTypes.string,
  size: PropTypes.number,
};

export default UsersAvatar;
