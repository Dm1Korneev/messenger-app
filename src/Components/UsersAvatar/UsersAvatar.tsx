import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import AccountIcon from '@material-ui/icons/AccountCircleOutlined';

type Props = {
  avatar?: string;
  author?: string;
  size?: number;
}

const UsersAvatar = ({ avatar, author, size }: Props) => {
  const style = size ? { width: size, height: size } : {};

  return (
    <Avatar variant="rounded" alt={author} src={avatar} style={style}>
      {!avatar && <AccountIcon fontSize="large" />}
    </Avatar>
  );
};

export default UsersAvatar;
