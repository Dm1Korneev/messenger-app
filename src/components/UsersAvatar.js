import React from "react";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import Avatar from "@material-ui/core/Avatar";
import withStyles from "@material-ui/core/styles/withStyles";

function UsersAvatar(props) {
  const { avatar, author, avatarHidden, classes } = props;

  return (
    <ListItemAvatar className={classes.ListItemAvatar}>
      <Avatar
        alt={author}
        src={avatar}
        className={avatarHidden ? classes.avatarHidden : ""}
      />
    </ListItemAvatar>
  );
}

const styles = theme => ({
  ListItemAvatar: {
    borderRadius: theme.spacing.unit / 2
  },
  avatarHidden: {
    visibility: "hidden"
  }
});

export default withStyles(styles)(UsersAvatar);
