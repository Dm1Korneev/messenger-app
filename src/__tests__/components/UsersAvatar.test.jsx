import React from 'react';

import UsersAvatar from 'Components/UsersAvatar';

jest.mock('@material-ui/core/ListItemAvatar', () => global.mockComponent('ListItemAvatar'));
jest.mock('@material-ui/core/Avatar', () => global.mockComponent('Avatar'));
jest.mock('@material-ui/icons/AccountCircleOutlined', () => global.mockComponent('AccountIcon'));

describe('render UsersAvatar component', () => {
  let wrapper;

  const props = {
    size: 60,
  };

  beforeAll(() => {
    wrapper = global.mount(
      <UsersAvatar {...props} />,
    );
  });

  test('component is render', () => {
    expect(wrapper.find('UsersAvatar').length).toBe(1);
  });

  test('ListItemAvatar subcomponent is render', () => {
    wrapper.update();
    expect(wrapper.find({ originalcomponent: 'ListItemAvatar' }).length).toBe(1);
  });

  test('Avatar subcomponent is render', () => {
    wrapper.update();
    expect(wrapper.find({ originalcomponent: 'Avatar' }).length).toBe(1);
  });

  test('AccountIcon subcomponent is render', () => {
    wrapper.update();
    expect(wrapper.find({ originalcomponent: 'AccountIcon' }).length).toBe(1);
  });
});

describe('snapshot-test UsersAvatar component', () => {
  test('Renders correct properties', () => {
    const props = {
      size: 60,
    };

    global.mountExpect(
      <UsersAvatar {...props} />,
    ).toMatchSnapshot();
  });
});
