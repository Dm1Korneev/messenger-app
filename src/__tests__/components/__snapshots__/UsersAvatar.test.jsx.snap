// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test UsersAvatar component Renders correct properties 1`] = `
<WithStyles(UsersAvatar)
  size={60}
>
  <UsersAvatar
    classes={
      Object {
        "ListItemAvatar": "UsersAvatar-ListItemAvatar-1",
      }
    }
    size={60}
  >
    <Component
      className="UsersAvatar-ListItemAvatar-1"
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    >
      <div
        originalcomponent="ListItemAvatar"
        testprops={
          Object {
            "children": <Unknown>
              <Unknown
                style={
                  Object {
                    "fontSize": 40,
                  }
                }
              />
            </Unknown>,
            "className": "UsersAvatar-ListItemAvatar-1",
            "style": Object {
              "height": 60,
              "width": 60,
            },
          }
        }
      >
        <Component>
          <div
            originalcomponent="Avatar"
            testprops={
              Object {
                "alt": undefined,
                "children": <Unknown
                  style={
                    Object {
                      "fontSize": 40,
                    }
                  }
                />,
                "src": undefined,
              }
            }
          >
            <Component
              style={
                Object {
                  "fontSize": 40,
                }
              }
            >
              <div
                originalcomponent="AccountIcon"
                testprops={
                  Object {
                    "style": Object {
                      "fontSize": 40,
                    },
                  }
                }
              />
            </Component>
          </div>
        </Component>
      </div>
    </Component>
  </UsersAvatar>
</WithStyles(UsersAvatar)>
`;
